pkg/ccl/backupccl/backup_test.go

```diff
@@ -1443,6 +1443,252 @@ into_db='restoredb', %s)`, encryptionOption), backupLoc1)
+			var holdUpCheckpointing bool
+			waitForProgress := make(chan struct{})
+			waitForPause := make(chan struct{})
+
+			params := base.TestClusterArgs{}
+			var mu syncutil.Mutex
+			knobs := base.TestingKnobs{
+				SQLExecutor: &sql.ExecutorTestingKnobs{
+					AfterBackupCheckpoint: func() {
+						mu.Lock()
+						if holdUpCheckpointing && len(exportedSpanCount) > 0 {
+							close(waitForProgress)
+							<-waitForPause
+							holdUpCheckpointing = false
+						}
+						mu.Unlock()
+					},
+				},
+				DistSQL: &execinfra.TestingKnobs{
+					BackupRestoreTestingKnobs: &sql.BackupRestoreTestingKnobs{
+						RunAfterExportingSpanEntry: func(ctx context.Context, response *kvpb.ExportResponse) {
+							mu.Lock()
+							if logExportSpanCount {
+								if response.Files != nil && len(response.Files) > 0 {
+									exportedSpan := response.Files[0].Span.String()
+									exportedSpanCount[exportedSpan] += 1
+								}
+							}
+							mu.Unlock()
+						},
+					},
+				},
+				JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
+			}
+			params.ServerArgs = base.TestServerArgs{Knobs: knobs}
+
+			tc, sqlDB, dir, cleanupFn := backupRestoreTestSetupWithParams(t, singleNode, 1,
+				InitManualReplication, params)
+			defer cleanupFn()
+
+			sqlDB.Exec(t, `CREATE DATABASE d`)
+			sqlDB.Exec(t, "USE d")
+
+			testCase.preBackupWorkload(ctx, t, sqlDB)
+
+			logExportSpanCount = true
+			holdUpCheckpointing = true
+			sqlDB.Exec(t, "SET CLUSTER SETTING bulkio.backup.checkpoint_interval = '10ms'")
+			var jobID int
+			sqlDB.QueryRow(t, testCase.backupCmd).Scan(&jobID)
+			<-waitForProgress
+			sqlDB.Exec(t, "PAUSE JOB $1", jobID)
+			jobutils.WaitForJobToPause(t, sqlDB, jobspb.JobID(jobID))
+			close(waitForPause)
+
+			backupDetails := jobutils.GetJobPayload(t, sqlDB, jobspb.JobID(jobID)).GetBackup()
+
+			mem, storage, _, _ := getDummyManifestInputs(ctx, t, backupDetails.URI, dir, tc.Servers[0].ExecutorConfig().(sql.ExecutorConfig))
+			desc, _, err := backupinfo.ReadBackupCheckpointManifest(ctx, mem, storage,
+				backupinfo.BackupManifestCheckpointName, nil, nil)
+			require.NoError(t, err)
+
+			// TODO(msbutler): Ideally, I'd assert that the manifest checkpoint
+			// contains at least one file, but it's very hard to control the order we
+			// receive export responses and the timing we flush the checkpoint.
+			// Further, if the last response contains actual data, it may end the
+			// backup flow before we're able to checkpoint it, so the test can't wait
+			// around for a nonempty checkpoint.
+			for _, file := range desc.Files {
+				require.Equal(t, testCase.checkpointsIntroducedSpans, fileFromIntroducedSpan(&file))
+			}
+
+			sqlDB.Exec(t, "RESUME JOB $1", jobID)
+			jobutils.WaitForJobToSucceed(t, sqlDB, jobspb.JobID(jobID))
+
+			for _, file := range desc.Files {
+				// Assert that all spans in the checkpoint were only exported once
+				require.Less(t, exportedSpanCount[file.Span.String()], 2)
```

Comment:
Are we using Less here because we aren't guaranteed to see every file?

Reply:
good catch. it should be Equal to 1. Fixed.